backend/internal/handlers/spahandler.go

```diff
@@ -0,0 +1,57 @@
+// As suggested by https://github.com/gorilla/mux#serving-single-page-applications
+
+package handlers
+
+import (
+	"net/http"
+	"os"
+	"path/filepath"
+)
+
+// SpaHandler implements the http.Handler interface, so we can use it
+// to respond to HTTP requests. The path to the static directory and
+// path to the index file within that static directory are used to
+// serve the SPA in the given static directory.
+type SpaHandler struct {
+	staticPath string
+	indexPath  string
+}
+
+// NewSpaHandler creates and returns a SpaHandler
+func NewSpaHandler(staticPath, indexPath string) SpaHandler {
+	return SpaHandler{staticPath: staticPath, indexPath: indexPath}
+}
+
+// ServeHTTP inspects the URL path to locate a file within the static dir
+// on the SPA handler. If a file is found, it will be served. If not, the
+// file located at the index path on the SPA handler will be served. This
+// is suitable behavior for serving an SPA (single page application).
+func (h SpaHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	// get the absolute path to prevent directory traversal
+	path, err := filepath.Abs(r.URL.Path)
+	if err != nil {
+		// if we failed to get the absolute path respond with a 400 bad request
+		// and stop
+		http.Error(w, err.Error(), http.StatusBadRequest)
```

Comment:
It's better not to leak any information about server file system via error messages. In this particular case it would be enough to return a generic error message ("Invalid request" for example) and log the actual error message.